howdoi/howdoi.py

```diff
@@ -621,7 +623,16 @@ def howdoi(raw_query):
         cache.set(cache_key, res)
     except (RequestsConnectionError, SSLError):
         res = {'error': f'Unable to reach {args["search_engine"]}. Do you need to use a proxy?\n'}
-
+    except BlockError:
+        BLOCKED_ENGINES.append(args['search_engine'])
+        if BLOCKED_ENGINES == SUPPORTED_SEARCH_ENGINES:
+            return howdoi(args)
```

Comment:
If these sets are now equal, which means that we've exhausted all the engines, when we want to return an error statement, right? The code as written today might cause an infinite loop.

Reply:
Yes, I tried doing just a `return` statement here, instead of `return howdoi(args)`, but that was throwing a pylint error (inconsistent return statements) because the rest of the return statements are returning stuff. Will look into this

Reply:
Yep, I thought that too, but added in the `return howdoi(args)` as simply `return` was throwing a pylint error. I've taken the alternate route with the `next()` function now. However, I'd like to test the code to see what happens when all search engines fail and whether it is the expected output (updated now). Any ideas on how to go about that (breaking stuff on purpose)?